posthog/src/main/java/com/posthog/internal/PostHogQueue.kt

```diff
@@ -257,8 +257,10 @@ internal class PostHogQueue(
     }
 
     private fun calculateDelay(retry: Boolean) {
-        val delay = if (retry) min(retryCount * retryDelaySeconds, maxRetryDelaySeconds) else config.flushIntervalSeconds
-        pausedUntil = config.dateProvider.addSecondsToCurrentDate(delay)
+        if (retry) {
+            val delay = min(retryCount * retryDelaySeconds, maxRetryDelaySeconds)
+            pausedUntil = config.dateProvider.addSecondsToCurrentDate(delay)
+        }
```

Comment:
I'm assuming the implicit return of pausedUntil isn't swallowed by the `if` block

(it's so long since I wrote any kotlin 🙈)

Reply:
do you mean:

```
if (!retry) return
val delay = min(retryCount * retryDelaySeconds, maxRetryDelaySeconds)
pausedUntil = config.dateProvider.addSecondsToCurrentDate(delay)
```

?

Reply:
discussed offline

Reply:
As discussed in Slack, I had missed this is a field and had completely parsed this incorrectly - so it did help me but not in the way I thought 🤣